### ISSUE TITLE
Make client message acknowledgement more fine grained

### ISSUE THREAD

**Issue by myronkscott** [Terracotta-OSS/terracotta-core#88]
Currently, the client only receives the completed ack once the passives and the actives have completed a requested action with the active waiting for passives to complete the action before starting.  It may be possible to run all actions in parallel on the actives and passives acknowledging back to the client that the active has finished the action as well as a second acknowledgement that the passives have completed the action.  This will give the client control as to which actions are important to it.

**Comment by jd0-sag** [comment]
Can that "active but not passive" ack be used, though?

We should run the active and passive, in parallel, but I don't know what other acks can expose meaning to the client.

Presumably, we still re-send the transaction, on restart/fail-over, unless it has completed everywhere, right?

**Comment by jd0-sag** [comment]
I am now working on this item.  I will solve the "parallel execution on active/passive" problem, which requires introduction of a "received" ACK to the active, for replicated messages.

I won't be exposing anything new to the clients, however, until we can determine what they would be and how anyone would use them.

This issue will be closed once active/passive parallel execution is working but any future work to expose these partial acknowledgements to the clients must be accompanied by a concrete example of why this would be required and how it would be interpreted.
